src/Microsoft.DotNet.ProjectModel/ProjectContextBuilder.cs

```diff
@@ -37,6 +37,8 @@ public class ProjectContextBuilder
 
         private Func<string, LockFile> LockFileResolver { get; set; }
 
+        private ProjectReader.Settings Settings { get; set; }
```

Comment:
@piotrpMSFT if this can wait, I wanted to relate this class with ProjectContext instead of reader (and hopefully remove that env variables although I already attempted to that cleanly, went too deep and gave up)